Prox/Prox/Utilities/PlaceUtilities.swift

```diff
@@ -95,12 +95,21 @@ struct PlaceUtilities {
             return
         }
 
-        // TODO: must cancel in-flight, or prevent updates to existing UI.
         let travelTimesResult = place.travelTimes(fromLocation: location)
         if !travelTimesResult.isFilled {
             view.setTravelTimesUIIsLoading(true)
         }
+
+        let idAtCallTime = place.id
```

Comment:
So, every object in the UIView heirarchy has an `restorationIdentifier` property that can be used to attach identifiers to views for the purposes or reuse or restoration. You could use that, so 
`(view as? UIView)?.restorationIdentifier = place.id` for example